### PR TITLE
[DOCS] Adds ML PRs to 6.6 release notes

### PR DESCRIPTION
**Pull request by lcawl**
This PR adds the machine learning PRs to the 6.6.0 Elasticsearch Release Notes.

### CODE REVIEW

**Comment by lcawl** [comment]
Need to https://github.com/elastic/elasticsearch/pull/36698 per discussion with @davidkyle

File: docs/reference/release-notes/6.6.asciidoc [hunk 1]
```diff
@@ -51,6 +51,34 @@ Also see <<breaking-changes-6.6,Breaking changes in 6.6>>.
...
+{pull}35387[#35387] (issue: {issue}35131[#35131])
+
+[float]
+[[enhancement-6.6.0]]
+=== Enhancements
+
+Machine Learning::
+* Create the {ml} annotations index {pull}36731[#36731] (issue: {issue}33376[#33376])
+* Split in batches and migrate all jobs and datafeeds {pull}36716[#36716]
+(issue: {issue}32905[#32905])
```
Comment (droberts195) [inline docs/reference/release-notes/6.6.asciidoc]: This one is really an implementation detail of the new feature "Store job configuration information in the new `.ml-config` index".  I'm not convinced it's worth listing separately.

File: docs/reference/release-notes/6.6.asciidoc [hunk 2]
```diff
@@ -59,4 +87,11 @@ Machine Learning::
...
 * Fix hang when closing a job or creating a forecast. This problem occurs if you
 created a forecast for a large job and temporary storage was not cleaned up.
-{ml-pull}352[#352] (issue: {ml-issue}350[#350])
\ No newline at end of file
+{ml-pull}352[#352] (issue: {ml-issue}350[#350])
+* Wait for autodetect to be ready in the datafeed {pull}37349[#37349]
+(issues: {issue}36810[#36810], {issue}37227[#37227])
+* Stop datafeeds when their jobs are stale {pull}37227[#37227]
+(issue: {issue}36810[#36810])
+* Fix x-pack usage regression caused by index migration {pull}36936[#36936]
```
Comment (droberts195) [inline docs/reference/release-notes/6.6.asciidoc]: This one was broken and fixed during 6.6 development, so has never been a problem in a released build.  Therefore I'd leave it out.

Comment (lcawl) [reply docs/reference/release-notes/6.6.asciidoc]: Ditto

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**